jwst/resample/resample_spec.py

```diff
@@ -313,6 +315,91 @@ def _max_virtual_slit_extent(self, wcs_list, target_ra, target_dec):
 
         return y_slit_min, y_slit_max
 
+    def build_miri_lrs_output(self):
+        """ Build MIRI LRS FIXEDSLIT output WCS. """
+        all_wavelength = []
+        all_ra_slit = []
+        all_dec_slit = []
+
+        for im, model in enumerate(self.input_models):
+            wcs = model.meta.wcs
+            bbox = wcs.bounding_box
+            grid = wcstools.grid_from_bounding_box(bbox)
+            ra, dec, lam = np.array(wcs(*grid))
+            # Handle vertical (MIRI) or horizontal (NIRSpec) dispersion.  The
+            # following 2 variables are 0 or 1, i.e. zero-indexed in x,y WCS order
+            spectral_axis = find_dispersion_axis(model)
+            spatial_axis = spectral_axis ^ 1
+
+            # Compute the wavelength array, trimming NaNs from the ends
+            # In many cases, a whole slice is NaNs, so ignore those warnings
+            warnings.simplefilter("ignore")
+            wavelength_array = np.nanmedian(lam, axis=spectral_axis)
+            warnings.resetwarnings()
+            wavelength_array = wavelength_array[~np.isnan(wavelength_array)]
+
+            all_wavelength = np.append(all_wavelength, wavelength_array)
+            this_minw = np.min(wavelength_array)
+            this_maxw = np.max(wavelength_array)
+            all_minw = np.min(all_wavelength)
+            all_maxw = np.max(all_wavelength)
+            if this_minw < all_minw:
+                addpts = wavelength_array[wavelength_array < all_minw]
+                all_wavelength = np.append(all_wavelength, addpts)
```

Comment:
If you want to keep the the wavelengths sorted, this should be modified:
```suggestion
                all_wavelength = np.append(addpts , all_wavelength)
```
**NOTE:** The above suggestion assumes wavelengths in `wavelength_array` are always increasing.